token/program-2022-test/tests/group_member_pointer.rs

```diff
@@ -0,0 +1,425 @@
+        .init_token_with_mint_keypair_and_freeze_authority(
+            mint,
+            vec![ExtensionInitializationParams::GroupMemberPointer {
+                authority: Some(*authority),
+                member_address,
+            }],
+            None,
+        )
+        .await
+        .unwrap();
+    context
+}
+
+#[tokio::test]
+async fn success_init() {
+    let payer = Keypair::new();
+    let group_mint = Keypair::new();
+    let group_update_authority = Keypair::new();
+    let member_mint = Keypair::new();
+    let member_authority = Keypair::new();
+
+    let program_test = setup_program_test();
+    let mut context = program_test.start_with_context().await;
+    context.set_account(
+        &payer.pubkey(),
+        &SolanaAccount {
+            lamports: 500_000_000,
+            ..SolanaAccount::default()
+        }
+        .into(),
+    );
+    let context = Arc::new(tokio::sync::Mutex::new(context));
+
+    let group_token = setup_group_mint(
+        context.clone(),
+        group_mint.insecure_clone(),
+        &group_update_authority.pubkey(),
+    )
+    .await
+    .token_context
+    .take()
+    .unwrap();
+
+    setup_token_group(
+        &group_token,
+        &group_token.mint_authority,
+        &group_update_authority.pubkey(),
+        &payer,
+    )
+    .await;
+
+    let member_token = setup_member_mint(
+        context,
+        member_mint.insecure_clone(),
+        &member_authority.pubkey(),
+    )
+    .await
+    .token_context
+    .take()
+    .unwrap()
+    .token;
+
+    let state = member_token.get_mint_info().await.unwrap();
+    assert!(state.base.is_initialized);
+    let extension = state.get_extension::<GroupMemberPointer>().unwrap();
+    assert_eq!(
+        extension.authority,
+        Some(member_authority.pubkey()).try_into().unwrap()
+    );
+    assert_eq!(
+        extension.member_address,
+        Some(member_mint.pubkey()).try_into().unwrap()
+    );
+}
+
+#[tokio::test]
+async fn fail_init() {
+    let payer = Keypair::new();
+    let group_mint = Keypair::new();
+    let group_update_authority = Keypair::new();
+
+    let program_test = setup_program_test();
+    let mut context = program_test.start_with_context().await;
+    context.set_account(
+        &payer.pubkey(),
+        &SolanaAccount {
+            lamports: 500_000_000,
+            ..SolanaAccount::default()
+        }
+        .into(),
+    );
+    let context = Arc::new(tokio::sync::Mutex::new(context));
+
+    let group_token = setup_group_mint(
+        context.clone(),
+        group_mint.insecure_clone(),
+        &group_update_authority.pubkey(),
+    )
+    .await
+    .token_context
+    .take()
+    .unwrap();
+
+    setup_token_group(
+        &group_token,
+        &group_token.mint_authority,
+        &group_update_authority.pubkey(),
+        &payer,
+    )
+    .await;
+
+    // fail with all none
+    let mut context = TestContext {
+        context,
+        token_context: None,
+    };
```

Comment:
nit: This can probably be simplified quite a bit too
```suggestion
    // fail with all none
    let mut context = TestContext::new().await;
```